lib/src/components/NxDropdown/__tests__/NxDropdown.test.tsx

```diff
@@ -5,357 +5,301 @@
+    const component = renderEl({ onToggleCollapse: jest.fn(), isOpen: true })!;
+
+    const escapeEvent = createEvent.keyDown(component, { key: 'Escape' });
+    const otherEvent = createEvent.keyDown(component, { key: 'Q' });
 
-    component.simulate('keyDown', { key: 'Escape', preventDefault: escPreventDefault });
-    component.simulate('keyDown', { key: 'Q', preventDefault: otherPreventDefault });
+    fireEvent(component, otherEvent);
+    expect(otherEvent.defaultPrevented).toBe(false);
 
-    expect(escPreventDefault).toHaveBeenCalled();
-    expect(otherPreventDefault).not.toHaveBeenCalled();
+    fireEvent(component, escapeEvent);
+    expect(escapeEvent.defaultPrevented).toBe(true);
   });
 
   it('does not call onToggleCollapse if ESC is pressed within the component when the dropdown is closed', function() {
-    const onToggleCollapse = jest.fn(),
-        component = getMountedComponent({ onToggleCollapse });
+    const onToggleCollapse = jest.fn();
+    const component = renderEl({ onToggleCollapse })!;
 
     expect(onToggleCollapse).not.toHaveBeenCalled();
-
-    component.simulate('keyDown', { key: 'Escape' });
+    fireEvent.keyDown(component, { key: 'Escape' });
     expect(onToggleCollapse).not.toHaveBeenCalled();
   });
 
   it('does not call onToggleCollapse if ESC is pressed within the component when the component is disabled',
       function() {
-        const onToggleCollapse = jest.fn(),
-            component = getMountedComponent({ onToggleCollapse, isOpen: true, disabled: true });
+        const onToggleCollapse = jest.fn();
+        const component = renderEl({ onToggleCollapse, isOpen: true, disabled: true })!;
 
         expect(onToggleCollapse).not.toHaveBeenCalled();
-
-        component.simulate('keyDown', { key: 'Escape' });
+        fireEvent.keyDown(component, { key: 'Escape' });
         expect(onToggleCollapse).not.toHaveBeenCalled();
       }
   );
 
   it('does not call onToggleCollapse if ESC is pressed within the component and onCloseKeyDown preventsDefault',
       function() {
-        const onToggleCollapse = jest.fn(),
-            component = getMountedComponent({
-              onToggleCollapse,
-              isOpen: true,
-              onCloseKeyDown: e => e.preventDefault()
-            }, { attachTo: container });
-
-        act(() => {
-          component.find('button.nx-dropdown__toggle').getDOMNode()
-              .dispatchEvent(new KeyboardEvent('keydown', { key: 'Escape', bubbles: true }));
-        });
-        component.update();
+        const onToggleCollapse = jest.fn();
+
+        const props: Partial<Props> = {
+          onToggleCollapse,
+          isOpen: true,
+          onCloseKeyDown: e => e.preventDefault()
+        };
+
+        const { rerender } = quickRender(props);
+
+        const button = screen.getByRole('button');
+        fireEvent.keyDown(button, { key: 'Escape', bubbles: true });
+
+        rerender(<NxDropdown {...minimalProps} {...props} />);
```

Comment:
Doesn't seem necessary

Reply:
> Doesn't seem necessary

Removed it here: 875cc55fa5870d28a090056472f32c6d5c2f90bb